Sources/ClientRuntime/EventStream/MessageMarshallable.swift

```diff
@@ -16,3 +16,16 @@ public protocol MessageMarshallable {
     /// - Returns: The marshalled `Message`.
```

Comment:
Defines a new typealias for closure that takes a stream struct, determines the specific instance of `Event`, and returns an `EventStream.Message` that has values "serialized" into it from the original `Event` for generic handling and encoding into Data.

A utility method `jsonMarshalClosure` is defined here that returns a closure which simply uses the old `marshall` method - that gets code-generated for each streaming union shape within the `MessageMarshallable` protocol conformance - to return an instance of `EventStream.Message`.